mkosi/resources/mkosi-tools/mkosi.prepare

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+# Copy the user-installed keyring from the host to the buildroot to avoid
+# failing signature verification. Do not overwrite existing files.
+if [ "$1" = "final" ] && [ -f /usr/share/keyrings/kali-archive-keyring.gpg ] && [ -d "$BUILDROOT/usr/share/keyrings" ]; then
+    cp -anv /usr/share/keyrings/kali-archive-keyring.gpg "$BUILDROOT/usr/share/keyrings/kali-archive-keyring.gpg"
+fi
```

Comment:
This should not be required, please drop it and rely on the package being installed instead.

Reply:
The package *is* installed on the host, but it is not available from Debian sources (or anywhere else than Kali itself). Thus the need to copy the file from the host. Otherwise the build fails because apt can not verify signatures without the keys, and then it reuses cached package lists without non-free-firmware.

I can probably remove it if it is ok to remove the microcode packages, too. But it would just be hiding a problem with the assumptions of the existing build logic.

Reply:
@bluca: I haven't found any. You can try it, but this won't solve the problem for Fedora and others. Links to sources are in the commit message.

Reply:
@mtdcr It's up to those distros to package the kali keyring if you want to build Kali images from those distros. Until that happens you'll need to be on a Kali host to build a Kali image or manually put the keyring in /usr/share/keyrings on the host machine used to build the Kali image. This is explicitly how mkosi chooses to operate, so please drop the prepare script.